torch/utils/cpp_extension.py

```diff
@@ -994,7 +994,9 @@ def CppExtension(name, sources, *args, **kwargs):
     libraries.append('c10')
     libraries.append('torch')
     libraries.append('torch_cpu')
-    libraries.append('torch_python')
+    if not kwargs.get('py_limited_api', False):
```

Comment:
From offline discussion, this flag is not actually related to the limited API. So let's keep it to "no torch python" or something along those line.

Reply:
Never mind, I don't agree anymore! :D It is related in that it is a documented/supported thing setuptools offers, so we should deal with this flag no matter what. https://github.com/pypa/setuptools/blob/9692cde009af4651819d18a1e839d3b6e3fcd77d/setuptools/extension.py#L127

We could offer _another_ flag to be explicit about not linking torch_python, but given that the intended `py_limited_api` doesn't work today, I'd prefer documenting that we're fixing that with explicit documentation that torch_python + its offered APIs will not be included. What do you think?